ethereum/core/src/test/java/org/hyperledger/besu/ethereum/bonsai/trielog/TrieLogLayerTests.java

```diff
@@ -0,0 +1,107 @@
+package org.hyperledger.besu.ethereum.bonsai.trielog;
+
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.ethereum.worldstate.StateTrieAccountValue;
+
+import java.util.Optional;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt256;
+import org.assertj.core.api.Assertions;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TrieLogLayerTests {
+  private TrieLogLayer trieLogLayer;
+  private TrieLogLayer otherTrieLogLayer;
+
+  @Before
+  public void setUp() {
+    trieLogLayer = new TrieLogLayer();
+    otherTrieLogLayer = new TrieLogLayer();
+  }
+
+  @Test
+  public void testAddAccountChange() {
+    Address address = Address.fromHexString("0x00");
+    StateTrieAccountValue oldValue = new StateTrieAccountValue(0, Wei.ZERO, Hash.EMPTY, Hash.EMPTY);
+    StateTrieAccountValue newValue =
+        new StateTrieAccountValue(1, Wei.fromEth(1), Hash.EMPTY, Hash.EMPTY);
+
+    Address otherAddress = Address.fromHexString("0x000000");
+    StateTrieAccountValue otherOldValue =
+        new StateTrieAccountValue(0, Wei.ZERO, Hash.EMPTY, Hash.EMPTY);
+    StateTrieAccountValue otherNewValue =
+        new StateTrieAccountValue(1, Wei.fromEth(1), Hash.EMPTY, Hash.EMPTY);
+
+    trieLogLayer.addAccountChange(address, oldValue, newValue);
+    otherTrieLogLayer.addAccountChange(otherAddress, otherOldValue, otherNewValue);
+
+    Assertions.assertThat(trieLogLayer).isEqualTo(otherTrieLogLayer);
+
+    Optional<StateTrieAccountValue> priorAccount = trieLogLayer.getPriorAccount(address);
+    Assertions.assertThat(priorAccount).isPresent();
+    Assertions.assertThat(priorAccount.get()).isEqualTo(oldValue);
+
+    Optional<StateTrieAccountValue> updatedAccount = trieLogLayer.getAccount(address);
+    Assertions.assertThat(updatedAccount).isPresent();
+    Assertions.assertThat(updatedAccount.get()).isEqualTo(newValue);
+  }
+
+  @Test
+  public void testAddCodeChange() {
+    Address address = Address.fromHexString("0xdeadbeef");
+    Bytes oldValue = Bytes.fromHexString("0x00");
+    Bytes newValue = Bytes.fromHexString("0x01030307");
+    Hash blockHash = Hash.fromHexStringLenient("0xfeedbeef02dead");
+
+    Address otherAddress = Address.fromHexString("0x0000deadbeef");
+    Bytes otherOldValue = Bytes.fromHexString("0x0000");
+    Bytes otherNewValue = Bytes.fromHexString("0x0001030307");
+    Hash otherBlockHash = Hash.fromHexStringLenient("0x00feedbeef02dead");
+
+    trieLogLayer.addCodeChange(address, oldValue, newValue, blockHash);
+    otherTrieLogLayer.addCodeChange(otherAddress, otherOldValue, otherNewValue, otherBlockHash);
```

Comment:
so is this testing that the leading zeros don't make a difference?

Reply:
I wanted an inexpensive way to ensure different objects are generated so we don't short-circuit the equals comparison.  the leading bytes don't matter for the address, but they do for the code bytes.  Fixed.

Reply:
weren't the tests passing before with the extra leading zeros though?     Assertions.assertThat(trieLogLayer).isEqualTo(otherTrieLogLayer);

Reply:
The failure was being masked in CI by spdx header failure for the new test class